### PR TITLE
[registrar] Detect more invalid characters in selectors so that we can report better errors. Fixes #55568.

### PR DESCRIPTION
**Pull request by rolfbjarne**
https://bugzilla.xamarin.com/show_bug.cgi?id=55568

### CODE REVIEW

File: src/ObjCRuntime/Registrar.cs [hunk 1]
```diff
@@ -186,7 +186,7 @@ public void VerifyRegisterAttribute (ref List<Exception> exceptions)
 				}
 			}
 
-			static char[] invalidSelectorCharacters = new char[] { ' ', '\t' };
+			static char[] invalidSelectorCharacters = new char[] { ' ', '\t', '?', '\\', '!', '|', '@', '"', '\'', '%', '&', '/', '(', ')', '=', '^', '[', ']', '{', '}', ',', '.', ';', '-', '\n' };
```
Comment (spouliot) [inline src/ObjCRuntime/Registrar.cs]: Should we make this public in the 'Selector' type ? Maybe a static method (not the array itself)

That would remove the duplication (with potential drift over time) and can be useful to others, both internally (designer) and externally

Comment (chamons) [inline src/ObjCRuntime/Registrar.cs]: If we don't do this, at least consider adding a comment pointing to the other duplicate in the test.

Comment (rolfbjarne) [inline src/ObjCRuntime/Registrar.cs]: The thing is that there's no official Objective-C specification, and this list of characters was created manually (by trying it out and see if it failed to build or not).

It's not meant to be a comprehensive list, just a best effort to help customers a little bit, so I'd prefer to not put this in our public API.

The list also varies depending on whether you're compiling Objective-C++ or Objective-C (Objective-C allows characters Objective-C++ doesn't).

Comment (rolfbjarne) [inline src/ObjCRuntime/Registrar.cs]: @spouliot @chamons I've added a comment to the source code about the duplicated information.

**Pull request merged**